doc/readme.md

```diff
@@ -0,0 +1,131 @@
+# TechTestApp
+
+[![Build Status][circleci-badge]][circleci]
+[![Release][release-badge]][release]
+[![GoReportCard][report-badge]][report]
+[![License][license-badge]][license]
+
+[circleci-badge]: https://circleci.com/gh/vibrato/TechTestApp.svg?style=shield&circle-token=8dfd03c6c2a5dc5555e2f1a84c36e33bc58ad0aa
+[circleci]: https://circleci.com/gh/vibrato/TechTestApp
+[release-badge]: http://img.shields.io/github/release/vibrato/TechTestApp/all.svg?style=flat
+[release]:https://github.com/vibrato/TechTestApp/releases
+[report-badge]: https://goreportcard.com/badge/github.com/vibrato/TechTestApp
+[report]: https://goreportcard.com/report/github.com/vibrato/TechTestApp
+[license-badge]: https://img.shields.io/github/license/vibrato/TechTestApp.svg?style=flat
+[license]: https://github.com/vibrato/TechTestApp/license
+
+The Vibrato techtest app is a golang application used for testing candidates applying to work with Vibrato.
+
+## Tech Test Application
+
+Single page application designed to be ran inside a container or on a vm (IaaS) with a postgres database to store data.
+
+It is completely self contained, and should not require any additional dependencies to run.
+
+## Install
+
+1. Download latest binary from release
+2. unzip into desired location
+3. and you should be good to go
+
+## Start server
+
+update `conf.toml` with database settings
+
+`./TechTestApp updatedb` to create a database and seed it with test data
+
+`./TechTestApp serve` will start serving requests
+
+## Interesting endpoints
+
+`/` - root endpoint that will load the SPA
+
+`/api/tasks/` - api endpoint to create, read, update, and delete tasks
+
+`/healthcheck/` - Used to validate the health of the application
+
+## Repository structure
+
+``` python
+.
+├── assets      # Asset directory for the application
+│   ├── css     # Contains all the css files for the web site
+│   ├── images  # Contains all the images for teh web site
+│   └── js      # Contains all the react javascript files
+├── cmd         # Command line UI logic is managed in this location
+├── config      # Contains the configuration logic for he application
+├── daemon      # Contains the logic of the daemon that runs and controll the app
+├── db          # Contains the data layet and db connectivity logic
+├── doc         # Documentation folder
+├── model       # Data model for the application
+├── scaffolds   # Scaffolds for deplying the application onto the cloud
+└── ui          # Web UI, routing, connectivity
+```
+
+## Application Architecture
+
+![architecture](images/architecture.png)
+
+The application itself is a React based single page application (SPA) with an API backend and a postgres database used for data persistence. It's been designed to be completely stateless and will deploy into most types of environments, be it container based or VM based.
+
+## Deploying into the cloud - scaffolds
```

Comment:
This is more an architectural point, but I'm not sure we should provide the scaffolding; We want to see how modularly they can build a solution, and also introduces too much complexity for them to get up and running. I reckon scrap it and let them provide a self-contained solution which we can spin up.